core/src/main/java/de/danoeh/antennapod/core/storage/PodDBAdapter.java

```diff
@@ -1363,17 +1282,10 @@ public final Cursor getFeedItemCursor(final String podcastUrl, final String epis
     public Cursor getImageAuthenticationCursor(final String imageUrl) {
         String downloadUrl = DatabaseUtils.sqlEscapeString(imageUrl);
         final String query = ""
-                + "SELECT " + KEY_USERNAME + "," + KEY_PASSWORD + " FROM " + TABLE_NAME_FEED_IMAGES
-                + " INNER JOIN " + TABLE_NAME_FEEDS
-                + " ON " + TABLE_NAME_FEED_IMAGES + "." + KEY_ID + "=" + TABLE_NAME_FEEDS + "." + KEY_IMAGE
-                + " WHERE " + TABLE_NAME_FEED_IMAGES + "." + KEY_DOWNLOAD_URL + "=" + downloadUrl
-                + " UNION SELECT " + KEY_USERNAME + "," + KEY_PASSWORD
-                + " FROM " + TABLE_NAME_FEED_IMAGES
-                + " INNER JOIN " + TABLE_NAME_FEED_ITEMS
-                + " ON " + TABLE_NAME_FEED_IMAGES + "." + KEY_ID + "=" + TABLE_NAME_FEED_ITEMS + "." + KEY_IMAGE
+                + "SELECT " + KEY_USERNAME + "," + KEY_PASSWORD + " FROM " + TABLE_NAME_FEED_ITEMS
                 + " INNER JOIN " + TABLE_NAME_FEEDS
-                + " ON " + TABLE_NAME_FEED_ITEMS + "." + KEY_FEED + "=" + TABLE_NAME_FEEDS + "." + KEY_ID
-                + " WHERE " + TABLE_NAME_FEED_IMAGES + "." + KEY_DOWNLOAD_URL + "=" + downloadUrl;
+                + " ON " + TABLE_NAME_FEED_ITEMS + "." + KEY_FEED + " = " + TABLE_NAME_FEEDS + "." + KEY_ID
```

Comment:
Regression Image authentication: the logic here supports images of `FeedItem`, but not the images of `Feed`.

Reply:
Nice catch. I used the same icon for feed and item in my tests, so I did not notice.

> but it'd be better if a test can be included with no external server dependency

Oh, I did not mean mean my server to be used for unit tests. Just for manual testing ;)